### PR TITLE
[An Cheng Yang] iP

### PR DESCRIPTION
**Pull request by anchengyang**
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Duke frees your mind by remembering things you need to do

It's 
- Text-based
- easy to learn
- ~~Fast~~ _SUPER FAST_ to use

## Steps to run
1. Pull the code from the repo [here](https://github.com/anchengyang/ip)
2. Run it
3. Add your tasks
4. Let it manage your tasks for you 👍

## Features
- [x] Managing todo
- [x] Managing deadlines
- [x] Managing events
- [x] Mark and unmark tasks
- [ ] Filter by date (coming soon)
- [ ] Filter by keyword (coming soon)

This is the main function:
   ```ruby
   public static void main(String[] args) {
        new Duke("tasks.txt").run();
    }
   ```

### CODE REVIEW

File: src/main/java/Deadline.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task{
```
Comment (salty-flower) [inline src/main/java/Deadline.java:1]: Perhaps it feels better to make this class part of some package?

File: src/main/java/Duke.java [hunk 2]
```diff
@@ -1,10 +1,174 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+enum Command {
+    TODO, DEADLINE, EVENT, DELETE, LIST, BYE, MARK, UNMARK;
```
Comment (salty-flower) [inline src/main/java/Duke.java:5]: I like your design to put commands within an enum class!

File: src/main/java/Duke.java [hunk 3]
```diff
@@ -1,10 +1,174 @@
...
+
+            try {
+                String[] words = input.split(" ");
+                Command command = Command.valueOf(words[0].toUpperCase());
+                if (command == Command.BYE) {
+                    System.out.println("    " + "Bye. Hope to see you again soon!");
+                    break;
+                }
+                switch(command) {
+                    case LIST:
```
Comment (salty-flower) [inline src/main/java/Duke.java:31]: Maybe have no indentation for case clauses here?

**Comment by anchengyang** [comment]
thanks for the comments @salty-flower

**Pull request closed without merging**